sahha-android/src/main/java/sdk/sahha/android/common/SahhaPermissions.kt

```diff
@@ -177,6 +177,11 @@ internal object SahhaPermissions : BroadcastReceiver() {
             manifestPermissions = Sahha.di.permissionManager.getManifestPermissions(context = context),
             sensors = sensors,
         ) { error, status, permissions ->
+            if (Sahha.di.permissionManager.isFirstHealthConnectRequest) {
+                callback?.invoke(null, SahhaSensorStatus.pending)
+                return@getTrimmedHcPermissions
+            }
+
             if (granted.containsAll(permissions)) {
```

Comment:
for future reference, might be safer to use an `else if` in case you forget to add a return